src/beforeStrip.php

```diff
@@ -1,107 +1,123 @@
-        $preprocessed=token_get_all($html);
-        
-        $HTML=array_filter($preprocessed, function($v) {
-            return is_array($v)&&$v[0]===T_INLINE_HTML;
+                                ) . '</doctypetag></div>';
+        $preprocessed = token_get_all($html);
+
+        $HTML = array_filter($preprocessed, function ($v) {
+            return is_array($v) && $v[0] === T_INLINE_HTML;
         });
-        $PHP=array_diff_key($preprocessed, $HTML);
+        $PHP = array_diff_key($preprocessed, $HTML);
         $this->init($HTML, $PHP, $html);
     }
-    
+
     protected function setMainTags(&$html)
     {
-        $doctypeOffset=stripos($html, '<!doctype');
-        $headOffset=stripos($html, '<head');
-        $htmlTagOffset=stripos($html, '<html');
-        $bodyOffset=stripos($html, '<body');
-        if (false!==$doctypeOffset) {
-            $endDoctypeOffset=strpos($html, '>', $doctypeOffset);
-            $this->doctype=substr($html, $doctypeOffset, $endDoctypeOffset-$doctypeOffset+1);
+        $doctypeOffset = stripos($html, '<!doctype');
+        $headOffset = stripos($html, '<head');
+        $htmlTagOffset = stripos($html, '<html');
+        $bodyOffset = stripos($html, '<body');
+        if (false !== $doctypeOffset) {
+            $endDoctypeOffset = strpos($html, '>', $doctypeOffset);
+            $this->doctype = substr($html, $doctypeOffset, $endDoctypeOffset - $doctypeOffset + 1);
         }
-        if (false!==$headOffset) {
-            $endHeadOffset=strpos($html, '>', $headOffset);
-            $this->head=substr($html, $headOffset, $endHeadOffset-$headOffset+1);
+        if (false !== $headOffset) {
+            $endHeadOffset = strpos($html, '>', $headOffset);
+            $this->head = substr($html, $headOffset, $endHeadOffset - $headOffset + 1);
         }
-        if (false!==$bodyOffset) {
-            $endBodyOffset=strpos($html, '>', $bodyOffset);
-            $this->body=substr($html, $bodyOffset, $endBodyOffset-$bodyOffset+1);
+        if (false !== $bodyOffset) {
+            $endBodyOffset = strpos($html, '>', $bodyOffset);
+            $this->body = substr($html, $bodyOffset, $endBodyOffset - $bodyOffset + 1);
         }
-        if (false!==$htmlTagOffset) {
-            $endHtmlTagOffset=strpos($html, '>', $htmlTagOffset);
-            $this->html_tag=substr($html, $htmlTagOffset, $endHtmlTagOffset-$htmlTagOffset+1);
+        if (false !== $htmlTagOffset) {
+            $endHtmlTagOffset = strpos($html, '>', $htmlTagOffset);
+            $this->html_tag = substr($html, $htmlTagOffset, $endHtmlTagOffset - $htmlTagOffset + 1);
         }
     }
+
     protected function init(&$HTML, &$PHP, &$html)
     {
-        list($is_h, $is_p)=array((bool)$HTML, (bool)$PHP);
-        list($this->is_html, $this->is_php, $this->multi, $this->content, $this->html, $this->php)=array($is_h, $is_p, $is_h&&$is_p, $html, $is_p ? $HTML : array(), $is_p ? $PHP : array());
```

Comment:
To be compatible with `php-7.0` syntax, these line of codes should be changed into variable assignment approach.